test/router.js

```diff
@@ -730,7 +730,7 @@ describe('Router', function () {
       .expect(200, 'saw POST /bar', cb)
     })
 
-    it('should match if path has trailing slash', function (done) {
+    it.skip('should match if path has trailing slash', function (done) {
```

Comment:
Ah, this one is an ongoing discussion but will likely be fixed in a 2.1 release, sorry - https://github.com/pillarjs/path-to-regexp/issues/123. The old one only passed because it always trimmed the trailing slash, the new one is leaving it there but I'm going to adjust the behaviour in non-ending mode to support this match behaviour.

Reply:
@dougwilson I could use your input though, does this feel valid to you. Building up a path, from looking at it, should be `/foo//` and `/foo//foo`, but in this mode it acts differently to written. I'm interested in the viewpoint in support of this oddity.

Reply:
I'll need to get a computer to understand the question, because mobile provides no way to view outside the diff to even see the path and url in the test.

Also thinking on this, i think we should just do a router 2 alpha now. Should we wait for 2.1.0 first or just put this pr back to 1.x and release a router 2 alpha 1 with it and follow with a alpha 2 after the 2.x regexp changes are settled?

Reply:
I am available this weekend to finish up what we need to get this merged into a 2 alpha, so if we could do that it would be awesome.  IMO getting this sooner will help people adjust better and give them more time to work with the new ways.

Reply:
Yea. That's why I'm wondering if we just start with the first alpha using path to regexp 1.x

Reply:
@blakeembrey How much is involved in putting together a PR for https://github.com/pillarjs/path-to-regexp/issues/123#issuecomment-335948313?  I haven't actually worked in `path-to-regxp` before, but if its simple enough I can take a stab at it before we decide what to do with this alpha.  If its too much, I can just revert this back to `1.7.0` simple enough.

Reply:
It should be very simple and I can implement it now, if that's what you need.

Reply:
It's mostly been a matter of figuring out how it _should work_. Right now, based on feedback and these tests, I'm thinking (using `/foo/` input):

* `end: true, strict: false` - `/foo//?`
* `end: true, strict: true` - `/foo/`
* `end: false, strict: false` - `/foo(?=/)`
* `end: false, strict: true` - `/foo/`

Reply:
Actually, comparing this one to https://github.com/pillarjs/path-to-regexp/issues/123, it's slightly different. Supporting https://github.com/pillarjs/path-to-regexp/issues/123 makes sense and is pretty straightforward. The reasoning behind supporting this one is less so. I'll wait for @dougwilson for feedback on why this style is supported since the `.use` is identical as using `/foo` here (it's specifically testing an edge case for this test).

Reply:
Ok, so I can see the entire test and it is testing that `app.use('/foo', fn)` runs `fn` for both `/foo` and `/foo/bar`. I think the reason this is tested and I continue to see many, many apps where they use the trailing slash in `app.use` is because it looks similar to `app.use('/', fn)` in that it ends with a slash. I'm not really sure which should be "right", but ideally with the amount of apps this would break, I wonder if it's worth all the turn over in everyone's apps.

Also: I'm describing the issue here correctly, right?

Reply:
Sounds correct, though I think you meant `app.use('/foo/', fn)`. With https://github.com/pillarjs/path-to-regexp/issues/123, it'll continue to work, but the child router won't pass in this example since it'll match as `/foo/` or `/` (whatever is input specifically) and the route trimmed will end up as `/foo/bar` -> `bar`. I definitely understand the potential churn here, but I've seen a few issues to the opposite effect which I'm trying to handle here - the unexpected behaviour of `'/'` becoming just `''` means you can't actually mount anything at specifically `/foo/` instead of `/foo`. Previous versions have gotten around the issue by saying "set `strict: true`", and if that's desired I can certainly do that.

Reply:
So are we just waiting to publish the `2.1.0` (or whatever you plan to version this as)?  Or is there anything else we need to get this pulled together?

Reply:
Yea, it sounds like to bump this to at least 2.1.0 (or back to 1.x and other as 2.x). I can merge whichever into the 2.0 branch that is going on.

Reply:
@dougwilson How important is this given it's a backward incompatible release of `path-to-regexp` and, I believe, it's a clearer implementation today. Here's a couple of related discussions: https://github.com/pillarjs/path-to-regexp/issues/120 and https://github.com/pillarjs/path-to-regexp/issues/123. With this change, the features in these issues would not be possible (specifically the trailing slash enforcement in https://github.com/pillarjs/path-to-regexp/issues/123, which larger worked by accident in 1.x).

Reply:
I mean, anything can change, I don't mind as long as we have migration instructions or at least a list of all the changes. Are those differences listed in the PR here already?

Of course Express continues to be num 1 framework for Node because the stability (in API, that is). The more differences the more it will invalidate all books and tutorials out there. It's not a blocker, only somethig to think about.

Reply:
Makes sense. I'd love to promote not using `app.use('/', fn)` as mentioned, since it's better overall just to skip the regexp (I believe this module handles it as a special case anyway). The only case it would break is the sub-router if it were trimming based on the matched `/`. However, even that I believe is handled with https://github.com/pillarjs/router/blob/ca2d51c145baa1bb4a3d4b55985b728560c328a1/index.js#L320. Let me do a 2.1 release and merge it into here, chances are it _just works_ because of these two handlers already built into the router that may have been handling 0.1.x edge-cases (for 1.x I had tried to bring that logic into `path-to-regexp`, for 2.x I realised pinning a ton of behaviour to Express.js edge cases might be a worse off idea).

Reply:
Done, `2.1.0` is released. When you're ready @wesleytodd, feel free to pull it in and see if it works as expected now. From looking at the module, I believe the edge-cases are all handled at the router level already.